runtime/testy/runtime.py

```diff
@@ -172,18 +185,29 @@ def runtimeTest():
             break
           spawnProcess(PROCESS_NAMES.STUDENT_CODE, runStudentCode, testName, 3)
           while True:
-            newBadThing = badThingsQueue.get(block=True)
-            print(newBadThing.event)
-            if newBadThing.event in restartEvents:
-              break
+            try:
+              newBadThing = badThingsQueue.get(block=True)
+              print(newBadThing.event)
+              if newBadThing.event in restartEvents:
+                break
+            except Exception as e:
+              print(e)
           stateQueue.put([SM_COMMANDS.RESET, []])
-          os.kill(allProcesses[PROCESS_NAMES.STUDENT_CODE].pid, signal.SIGKILL)
+          student_process = allProcesses[PROCESS_NAMES.STUDENT_CODE]
+          student_process.terminate()
+          time.sleep(.01) # Give the OS a chance to terminate the other process
+          if student_process.is_alive():
```

Comment:
This code is a repeat of above, we should just have a function that handles this, even if it is not ideal.